app/controllers/service_provider_controller.rb

```diff
@@ -2,11 +2,27 @@ class ServiceProviderController < ApplicationController
   protect_from_forgery with: :null_session
 
   def update
-    if FeatureManagement.use_dashboard_service_providers?
-      ServiceProviderUpdater.new.run
-      SecureHeadersWhitelister.new.run
+    authorize do
+      if FeatureManagement.use_dashboard_service_providers?
+        ServiceProviderUpdater.new.run
+        SecureHeadersWhitelister.new.run
+      end
+
+      render json: { status: 'If the feature is enabled, service providers have been updated.' }
+    end
+  end
+
+  private
+
+  def authorize
+    if authorization_token == Figaro.env.dashboard_api_token
+      yield
+    else
+      render nothing: true, status: 401
     end
+  end
 
-    render json: { status: 'If the feature is enabled, service providers have been updated.' }
+  def authorization_token
+    request.headers['X-LOGIN-DASHBOARD-TOKEN']
```

Comment:
This is not about how to name our header, if our canonical header name is `X-LOGIN-DASHBOARD-TOKEN` that's 100% fine, not here to argue that, but we will need to change our code to read it as `HTTP_X_LOGIN_DASHBOARD_TOKEN`, otherwise we won't be able to read it when we're serving web requests.